src/plugins/intel_cpu/src/emitters/x64/jit_emitter.hpp

```diff
@@ -130,6 +132,74 @@ class jit_emitter : public ov::snippets::Emitter {
         }
     }
 
+    // below 4 functions must be inline funtions to avoid corrupted rsp by function call, so defined inside class declaration.
+    void internal_call_preamble() const {
```

Comment:
Is there a reason why these implementations are in the header? Let's move them to the cpp file to keep class definitions more nice and readable.

Reply:
As note in https://github.com/openvinotoolkit/openvino/pull/19493#discussion_r1386708413, this function must be inline.  Define it in class declaration means it's inline implicitely. If move definition in cpp, we need explicit qualifier `inline`, but for compiler, this inline is just suggestion, compiler could ignore it. So we need a way to force inline it. maybe something like this is needed.

> #ifdef _MSC_VER_ // for MSVC
> #define forceinline __forceinline
> #elif defined __GNUC__ // for gcc on Linux/Apple OS X
> #define forceinline __inline__ __attribute__((always_inline))
> #else
> #define forceinline
> #endif

Reply:
discussed offline, not force inline is needed, and implementation is moved to cpp